### ISSUE TITLE
Configurable Retry Policy

### ISSUE THREAD

**Issue by joe-maley** [Azure/azure-storage-cpplite#61]
Both `class blob_client` constructors initialize `executor_context blob_client::m_context` with the default `class retry_policy`. The default retry policy retries 26 times with an exponential back-off up to 60 seconds on each retry.

Please support a configurable retry policy. I'd suggest:
1. Allow `class retry_policy` to be constructed with both a maximum retry count and a retry interval.
2. Allow `class blob_client` to be constructed with an instance of `class retry_policy`.

**Comment by JinmingHu-MSFT** [comment]
@joe-maley Yes, you're right. The existing retry policy doesn't make too much sense. We also have other customers complaining about this, and also have pending PR https://github.com/Azure/azure-storage-cpplite/pull/54 trying to fix this problem. We'll consider adding this feature in future release. Thanks for your feedback.

**Comment by JinmingHu-MSFT** [comment]
We added configurable retry policy in latest 0.3.0 release. The default retry count is also changed from 25 to 3.
